include/inviwo/core/io/datareaderfactory.h

```diff
@@ -118,14 +118,22 @@ std::vector<FileExtension> DataReaderFactory::getExtensionsForType() const {
 template <typename T>
 std::unique_ptr<DataReaderType<T>> DataReaderFactory::getReaderForTypeAndExtension(
     std::string_view path) const {
-    for (auto& elem : map_) {
-        if (util::iCaseEndsWith(path, elem.first.extension_)) {
-            if (auto r = dynamic_cast<DataReaderType<T>*>(elem.second)) {
-                return std::unique_ptr<DataReaderType<T>>(r->clone());
+    std::vector<std::pair<size_t, DataReaderType<T>*>> candidates;
+    for (auto& [ext, reader] : map_) {
+        if (util::iCaseEndsWith(path, ext.extension_)) {
+            if (auto r = dynamic_cast<DataReaderType<T>*>(reader)) {
+                candidates.emplace_back(ext.extension_.size(), r);
             }
         }
     }
-    return std::unique_ptr<DataReaderType<T>>{};
+
+    if (auto it = std::max_element(candidates.begin(), candidates.end(),
```

Comment:
Might want to copy the removed comment from above.
```suggestion
    // Sort extensions by length to first compare long extensions. Avoids selecting extension xyz in
    // case xyzw exist, see getReaderForTypeAndExtension
    if (auto it = std::max_element(candidates.begin(), candidates.end(),
```